src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -1303,6 +1306,12 @@ private bool ShouldCache(long number)
             blockNumber ??= _headerStore.GetBlockNumber(blockHash);
             if (blockNumber is not null)
             {
+                // Although thread-safe, because the blocks are so large it causes a lot of contention
+                // on the allocator inside RocksDb; which then impacts block processing heavily, especially on Windows.
+                // We take the lock here instead to reduce contention on the allocator in the db.
+                // A better solution would be to change the allocator https://github.com/NethermindEth/nethermind/issues/6107
+                using var handle = _allocatorLock.Acquire();
+
                 block = _blockStore.Get(
```

Comment:
Probably should be in BlockStore. There is an extension method that also does the deserialization. Probably pass the lock into the method so that the deserialization is not locked.

Reply:
Will have allocated and deallocted returning the memory to the pool to be used for next block at this level? So also use less memory and not contend on free?